src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextWriter.cs

```diff
@@ -171,7 +171,7 @@ private void WritePortableTarget(string key, IReadOnlyList<RuntimeLibrary> runti
                     Debug.Assert(compilationLibrary.RuntimeStoreManifestName == null);
                 }
 
-                Library library = (Library)compilationLibrary ?? (Library)runtimeLibrary;
+                Library library = (Library?)compilationLibrary ?? runtimeLibrary ?? throw new NullReferenceException(nameof(library));
```

Comment:
We shouldn't ever throw `NullReferenceException` explicitly. Just add a `!` here for this PR.